flytekit/core/promise.py

```diff
@@ -99,6 +99,10 @@ def my_wf(in1: int, in2: int) -> int:
         try:
             if type(v) is Promise:
                 v = await resolve_attr_path_in_promise(v)
+            elif type(v) is list:
```

Comment:
Should we use recursive here? What if we have a nested list.

Reply:
You are correct